cdap-ui/app/tracker/templates/preview.html

```diff
@@ -59,3 +59,10 @@
   ng-if="PreviewController.previewData.length === 0 && !PreviewController.loading">
   <p>No data</p>
```

Comment:
Preview message will be shown only when **previewNotAvailable** gets true and when it gets true, this  means error occurred while executing query and hence **previewData** will be undefined and its length won't be equal to zero. 

On the other hand **previewNotAvailable** wont get true if query is successful and even when we have empty data.